src/presenter/Trip-presenter.js

```diff
@@ -48,59 +117,83 @@ export default class TripPresenter {
       return;
     }
 
-    this.#sortPoints(sortType);
-    this.#clearPointList();
-    this.#renderPoints();
-  }
-
-  #sortPoints = (sortType) => {
-    switch (sortType) {
-      case SortType.PRICE.text:
-        this.#boardPoints.sort(sortPointsByPrice);
-        break;
-      case SortType.TIME.text:
-        this.#boardPoints.sort(sortPointsByTime);
-        break;
-      default:
-        this.#boardPoints = [...this.#sourceBoardPoints];
-    }
-
     this.#currentSortType = sortType;
+    // this.#clearPointList();
+    // this.#renderPoints(this.points);
+    this.#clearBoard();
+    this.#renderBoard();
   }
 
+  // #sortPoints = (sortType) => {
+  //   switch (sortType) {
+  //     case SortType.PRICE.text:
+  //       this.#boardPoints.sort(sortPointsByPrice);
+  //       break;
+  //     case SortType.TIME.text:
+  //       this.#boardPoints.sort(sortPointsByTime);
+  //       break;
+  //     default:
+  //       this.#boardPoints = [...this.#sourceBoardPoints];
+  //   }
+
+  //   this.#currentSortType = sortType;
+  // }
+
   #renderPoint = (point) => {
-    const pointPresenter = new PointPresenter(this.#pointListComponent, this.#handlePointChange, this.#handleModeChange);
+    const pointPresenter = new PointPresenter(this.#pointListComponent, this.#handleViewAction, this.#handleModeChange);
     pointPresenter.init(point);
     this.#pointPresenter.set(point.id, pointPresenter);
   };
 
-  #renderPoints = () => {
-    this.#boardPoints
-      .forEach((boardPoint) => this.#renderPoint(boardPoint));
-  }
-
-  #clearPointList = () => {
-    this.#pointPresenter.forEach((presenter) => presenter.destroy());
-    this.#pointPresenter.clear();
+  #renderPoints = (points) => {
+    points.forEach((point) => this.#renderPoint(point));
   }
 
   #renderSort = () => {
+    this.#sortComponent = new SortView(this.#currentSortType);
+
     render(this.#tripContainer, this.#sortComponent, renderPosition.AFTERBEGIN);
-    this.#sortComponent.setSortChengeClickHandler(this.#handleSortTypeChange);
+    this.#sortComponent.setSortChangeClickHandler(this.#handleSortTypeChange);
   }
 
 
   #renderNoPoints = () => {
+    this.#noPointsComponent = new MessageWithoutPointsView(this.#filterType);
     render(this.#tripContainer, this.#noPointsComponent, renderPosition.BEFOREEND);
   }
 
+  #clearBoard = ({ resetRenderedTotalPrice = false, resetSortType = false } = {}) => {
+    const totalPointsPrice = 0;
+    //this.points.price.forEach((sum, price) => sum + price);
```

Comment:
`const totalPointsPrice = this.points.reduce((sum, point) => sum + point.price, 0);`